Core/include/Acts/TrackFitting/detail/GsfActor.hpp

```diff
@@ -179,14 +179,14 @@ struct GsfActor {
     // Prints some VERBOSE things and performs some asserts. Can be removed
     // without change of behaviour
     const detail::ScopedGsfInfoPrinterAndChecker printer(state, stepper,
-                                                         logger());
+                                                         navigator, logger());
 
     // We only need to do something if we are on a surface
-    if (not state.navigation.currentSurface) {
+    if (not navigator.currentSurface(state.navigation)) {
       return;
     }
 
-    const auto& surface = *state.navigation.currentSurface;
+    const auto& surface = *navigator.currentSurface(state.navigation);
```

Comment:
I actually would prefer propagating this reference through to the other other methods, and not the navigator type, but this of course is a cosmetic detail.

Reply:
makes sense! I fear that might trigger a few more design choices like with `evaluateMaterialSlab`. I would argue that the current state of the refactor is not changing the design but just refactors the access for the navigation state. therefor I would rather use a follow up PR to fix the design